torch/utils/benchmark/utils/compile.py

```diff
@@ -0,0 +1,175 @@
+import torch
+
+__all__ = ["bench_all", "benchmark_compile"]
+
+import torch._dynamo
+from torch.utils.benchmark import Timer
+
+from typing import Optional, List, Callable, Union, Any, cast
+
+try:
+    from tabulate import tabulate
+    HAS_TABULATE = True
+except ImportError:
+    HAS_TABULATE = False
+
+_warned_tensor_cores = False
+_default_float_32_precision = torch.get_float32_matmul_precision()
+
+
+if HAS_TABULATE:
+    def _enable_tensor_cores():
+        global _warned_tensor_cores
+
+        if torch.cuda.is_available():
+            if torch.backends.cuda.matmul.allow_tf32 is False and torch.cuda.get_device_capability() >= (8, 0):
+                torch.set_float32_matmul_precision("high")
```

Comment:
I have a noob question: why is the effect of tensor cores not proportional across all backends? E.g., if I'm comparing eager and inductor, as long as I compare them consistently with or without tensor cores, I feel like I should get the same factor? Does tensorcores change where the bottlenecks are?

Reply:
If the end user isn't using tensor cores, it seems disingenous to give benchmark numbers with tensor cores, and then they will proceed to not use it.

Reply:
Tensorcores move bottlenecks from matmul (that inductor doesn't help with) to other, bw-bound, ops that inductor fuses. E.g. w/o tensorcores 90% of eager workload can be in matmuls, so if inductor accelerates remaining 10% by 2x, it's 5% e2e, with tensorcores matmuls would be say 60%, so accelerating remainder by 2x is 20% e2e.

Reply:
alright, thanks